pkg/k8s/k8s_pod.go

```diff
@@ -410,82 +417,118 @@ func buildResources(memoryRequest string, memoryLimit string, cpuRequest string)
-	env := spec.Env
-	volumes := spec.Volumes
-	files := spec.Files
-
+// prepareContainer creates a v1.Container from a given ContainerConfig.
+func prepareContainer(config ContainerConfig) (v1.Container, error) {
 	// Build environment variables from the given map
-	podEnv := buildEnv(env)
+	podEnv := buildEnv(config.Env)
 
-	// Build pod volumes from the given map
-	podVolumes, err := buildPodVolumes(name, len(volumes), len(files))
+	// Build container volumes from the given map
+	containerVolumes, err := buildContainerVolumes(config.Name, config.Volumes, config.Files)
 	if err != nil {
-		return v1.PodSpec{}, fmt.Errorf("failed to build pod volumes: %v", err)
+		return v1.Container{}, fmt.Errorf("failed to build container volumes: %v", err)
 	}
 
-	// Build container volumes from the given map
-	containerVolumes, err := buildContainerVolumes(name, volumes, files)
+	resources, err := buildResources(config.MemoryRequest, config.MemoryLimit, config.CPURequest)
 	if err != nil {
-		return v1.PodSpec{}, fmt.Errorf("failed to build container volumes: %v", err)
+		return v1.Container{}, fmt.Errorf("failed to build resources: %v", err)
+	}
+
+	return v1.Container{
+		Name:           config.Name,
+		Image:          config.Image,
+		Command:        config.Command,
+		Args:           config.Args,
+		Env:            podEnv,
+		VolumeMounts:   containerVolumes,
+		Resources:      resources,
+		LivenessProbe:  config.LivenessProbe,
+		ReadinessProbe: config.ReadinessProbe,
+		StartupProbe:   config.StartupProbe,
+	}, nil
+}
+
+// prepareInitContainers creates a slice of v1.Container as init containers.
+func prepareInitContainers(config ContainerConfig, init bool) ([]v1.Container, error) {
+	if !init || len(config.Volumes) == 0 {
+		return nil, nil
 	}
 
-	var initContainers []v1.Container
-	if len(volumes) > 0 && init {
-		// Build init containers volumes and command from the given map
-		initContainerVolumes, err := buildInitContainerVolumes(name, volumes)
-		if err != nil {
-			return v1.PodSpec{}, fmt.Errorf("failed to build init container volumes: %v", err)
-		}
-		initContainerCommand, err := buildInitContainerCommand(name, volumes)
-		if err != nil {
-			return v1.PodSpec{}, fmt.Errorf("failed to build init container command: %v", err)
-		}
+	initContainerVolumes, err := buildInitContainerVolumes(config.Name, config.Volumes)
+	if err != nil {
+		return nil, fmt.Errorf("failed to build init container volumes: %v", err)
+	}
+	initContainerCommand, err := buildInitContainerCommand(config.Name, config.Volumes)
+	if err != nil {
+		return nil, fmt.Errorf("failed to build init container command: %v", err)
+	}
 
-		user := int64(0)
+	user := int64(0)
 
-		initContainers = []v1.Container{
-			{
-				Name:  "volume-whatever",
-				Image: image,
-				SecurityContext: &v1.SecurityContext{
-					RunAsUser: &user,
-				},
-				Command:      initContainerCommand,
-				VolumeMounts: initContainerVolumes,
+	return []v1.Container{
+		{
+			Name:  "volume-whatever",
```

Comment:
the name is okay or should be:
```suggestion
			Name:  config.Name,
```